### PR TITLE
Feat: get blogs by Id

### PR DESCRIPTION
**Pull request by Nifilat**
​

## Description

Added functionality for retrieving blog posts by id. This includes creating a new endpoint /api/v1/blogs/{id} that fetches
blog posts that match the id from the database. The endpoint returns a single blog post, containing details such as title, content, author, tags, etc.

​

## Related Issue (Link to issue ticket)

[#236 ](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/236)
​

## Motivation and Context

​This change is required to provide an endpoint for clients to retrieve a blog post by its id. It enhances the application's functionality by allowing users to view a single blog post.

## How Has This Been Tested?

The tests were carried out using pytest.
I created a mock blog and was able to fetch the details successfully. 
I also tested for a successful 404 response for IDs that do not exist

## Screenshots (if appropriate - Postman, etc):
Screenshot 2024-07-24 at 22 12 01
​

## Swagger Documentation
Screenshot 2024-07-24 at 22 08 13

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Blog id is supposed to be a string not uuid.

File: api/v1/routes/blog.py [hunk 1]
```diff
@@ -15,3 +19,11 @@ def get_all_blogs(db: Session = Depends(get_db)):
...
     return blogs
 
+
+@blog.get("/{id}", response_model=SingleResponse)
+def get_blog_by_id(id: str, db: Session = Depends(get_db)):
+    blog_post = blog_service.fetch(db, id)
+    if not blog_post:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="Blog post not found.")
+    return blog_post
```
Comment (zxenonx) [inline api/v1/routes/blog.py:29]: wrong response format

File: api/v1/schemas/blog.py [hunk 2]
```diff
@@ -16,5 +17,20 @@ class BlogResponse(BaseModel):
     updated_at: datetime
 
     class Config:
-        orm_mode = True
+        from_attributes = True
+
+
+class SingleResponse(BaseModel):
```
Comment (zxenonx) [inline api/v1/schemas/blog.py:23]:
```suggestion
class BlogPostResponse(BaseModel):
```

**Review by zxenonx — CHANGES_REQUESTED** [review]
Thanks @Nifilat 
I left some comments.

File: api/v1/routes/blog.py (hunk 1 above)
Comment (Nifilat) [reply api/v1/routes/blog.py:29]: I will work on that now, thank you

Comment (Nifilat) [reply api/v1/routes/blog.py:37]: @zxenonx ... I have implemented the change the correct the response format.
Thank you.

**Comment by buka4rill** [comment]
wrong PR and commit naming. Closed!

**Pull request closed without merging**